aggregator/src/binaries/janus_cli.rs

```diff
@@ -91,6 +182,101 @@ impl Command {
         // function with the main command logic.
         let kube_client = LazyKubeClient::new();
         match self {
+            Command::GenerateGlobalHpkeKey {
+                kubernetes_secret_options,
+                id,
+                kem,
+                kdf,
+                aead,
+                hpke_config_out_file,
+            } => {
+                let datastore = datastore_from_opts(
+                    kubernetes_secret_options,
+                    command_line_options,
+                    config_file,
+                    &kube_client,
+                )
+                .await?;
+
+                generate_global_hpke_key(
+                    &datastore,
+                    command_line_options.dry_run,
+                    (*id).into(),
+                    (*kem).into(),
+                    (*kdf).into(),
+                    (*aead).into(),
+                    hpke_config_out_file.as_deref(),
+                )
+                .await
+            }
+
+            Command::SetGlobalHpkeKeyState {
+                kubernetes_secret_options,
+                id,
+                state,
+            } => {
+                let datastore = datastore_from_opts(
+                    kubernetes_secret_options,
+                    command_line_options,
+                    config_file,
+                    &kube_client,
+                )
+                .await?;
+
+                set_global_hpke_key_state(
+                    &datastore,
+                    command_line_options.dry_run,
+                    (*id).into(),
+                    *state,
+                )
+                .await
+            }
+
+            Command::AddTaskprovPeerAggregator {
+                kubernetes_secret_options,
+                peer_endpoint,
+                role,
+                verify_key_init,
+                collector_hpke_config_file,
+                report_expiry_age_secs,
+                tolerable_clock_skew_secs,
+                aggregator_auth_token,
+                collector_auth_token,
+            } => {
+                let datastore = datastore_from_opts(
+                    kubernetes_secret_options,
+                    command_line_options,
+                    config_file,
+                    &kube_client,
+                )
+                .await?;
+
+                // Parse flags into proper types.
+                let verify_key_init =
+                    VerifyKeyInit::try_from(URL_SAFE_NO_PAD.decode(verify_key_init)?.as_ref())?;
```

Comment:
nit: we can make an `impl FromStr for VerifyKeyInit` with this logic, similar to what we have in messages, then clap can directly parse a VerifyKeyInit.

I generally like to coerce clap into using a ValueParser https://docs.rs/clap/latest/clap/macro.value_parser.html, which in practice has usually involved implementing `FromStr`. This makes clap print pretty error messages, and generally `FromStr` is nice to have around elsewhere.

Same idea with the bearer tokens below. Can take it or leave it though, it's a minor refactor.

Reply:
Hmm, I like this idea. Implementing on `VerifyKeyInit` seems like an obvious win as there is already serde code that uses the same format. `AuthenticationToken` was less obvious; I considered wrapping a newtype, but I decided I'm OK adding a `FromStr` as there was no string format for an `AuthenticationToken` yet, and prefixed-value is flexible enough to handle any new token types that may come to exist.